tests/keras/backend/backend_test.py

```diff
@@ -248,21 +258,55 @@ def test_random_variables(self):
                                       shape_or_val=False,
                                       assert_value_equality=False)
 
-    @pytest.mark.skipif(K.backend() != 'tensorflow', reason='Not supported.')
     def test_batch_dot_shape(self):
-        x_batch = K.ones(shape=(32, 20))
-        y_batch = K.ones(shape=(32, 20))
-        xy_batch_dot = K.batch_dot(x_batch, y_batch, axes=1)
-        assert_allclose(K.eval(xy_batch_dot), np.ones((32, 1)) * 20, atol=1e-05)
-        xy_batch_dot = K.batch_dot(x_batch, y_batch, axes=0)
-        assert_allclose(K.eval(xy_batch_dot), np.ones((20, 1)) * 32, atol=1e-05)
-        # making sure swapping axes when ndim == 2 works
-        x_batch = K.ones(shape=(32, 20))
-        y_batch = K.ones(shape=(20, 32))
-        xy_batch_dot = K.batch_dot(x_batch, y_batch, axes=(0, 1))
-        assert_allclose(K.eval(xy_batch_dot), np.ones((20, 1)) * 32, atol=1e-05)
-        xy_batch_dot = K.batch_dot(x_batch, y_batch, axes=(1, 0))
-        assert_allclose(K.eval(xy_batch_dot), np.ones((32, 1)) * 20, atol=1e-05)
+        # Note : batch_dot implementation is different for
+        # placeholders and variables in CNTK backend
+
+        test_cases = []
+        test_cases.append([(None, 3, 4, 5), (None, 2, 3, 4), (2, 3),
+                           (None, 3, 5, 2, 3)])
+        test_cases.append([(None, 3, 4, 5), (None, 2, 4), 2, (None, 3, 5, 2)])
+        test_cases.append([(None, 3, 4), (None, 2, 3, 4), (2, 3), (None, 3, 2, 3)])
+        test_cases.append([(None, 4), (None, 3, 4), (1, 2), (None, 3)])
+        test_cases.append([(None, 4), (None, 4), None, (None, 1)])
+
+        batch_size = 7
+
+        def batch_shape(shape):
+            return (batch_size, ) + shape[1:]
+
+        def ones(shape):
+            return np.ones(batch_shape(shape))
```

Comment:
Maybe should we use `np.random.uniform` and compare with the output of the `batch_dot` in `reference_operations.py` rather than hard coding the solution?